services/galley/src/Galley/API/Update.hs

```diff
@@ -105,13 +113,114 @@ unblockConv (usr ::: conn ::: cnv) = do
     conv' <- acceptOne2One usr conv conn
     setStatus status200 . json <$> conversationView usr conv'
 
-joinConversation :: UserId ::: ConnId ::: ConvId ::: JSON -> Galley Response
-joinConversation (zusr ::: zcon ::: cnv ::: _) = do
+updateConversationAccess :: UserId ::: ConnId ::: ConvId ::: Request ::: JSON -> Galley Response
+updateConversationAccess (usr ::: zcon ::: cnv ::: req ::: _ ) = do
+    body <- fromBody req invalidPayload
+    let targetAccess = Set.fromList (toList (cupAccess body))
```

Comment:
The semantics of the access mode update endpoint are that clients should specify the full set of access modes, right?
In that case, does an access mode of `Invite | TeamOnly | Code` make sense, for instance?

Reply:
> The semantics of the access mode update endpoint are that clients should specify the full set of access modes, right?
Yes

> In that case, does an access mode of Invite | TeamOnly | Code make sense, for instance?
Well, it's `team`, which is meant to mean "team members can join". It's not "team-only".

Reply:
Right, my bad there. But so wouldn't it make sense to have a conversation access as `Invite | Team` that would also cause non-team members to be kicked out? Or by having a `Team` access only we mean that users can still be added to the conversation as long as they are part of the same team?